features/src/autogluon/features/generators/datetime.py

```diff
@@ -39,42 +39,54 @@ def _transform(self, X: DataFrame) -> DataFrame:
+        holidays_list = holidays.CountryHoliday(country)
         for datetime_feature in self.features_in:
-            # TODO: Be aware: When converted to float32 by downstream models, the seconds value will be up to 3 seconds off the true time due to rounding error. If seconds matter, find a separate way to generate (Possibly subtract smallest datetime from all values).
-            # TODO: could also return an extra boolean column is_nan which could provide predictive signal.
-            X_datetime[datetime_feature] = pd.to_datetime(X[datetime_feature], errors='coerce').fillna(self._fillna_map[datetime_feature])
-            # X_datetime[datetime_feature] = pd.to_timedelta(X_datetime[datetime_feature]).dt.total_seconds()
-            # Parse the date into lots of derived fields.
-            # Most of the pandas Series.dt properties are here, a few are omitted (e.g. is_month_start) if they can be inferred
-            # from other features.
-            for feature in self.features:
-                X_datetime[datetime_feature + '.' + feature] = getattr(X_datetime[datetime_feature].dt, feature).astype(int)
-
-            X_datetime[datetime_feature] = pd.to_numeric(X_datetime[datetime_feature])
-
+            if type(X[datetime_feature].iloc[0]) == pd.Timestamp:
+                X_datetime[datetime_feature + "_holiday"] = 0
+                # TODO: Be aware: When converted to float32 by downstream models, the seconds value will be up to 3 seconds off the true time due to rounding error. If seconds matter, find a separate way to generate (Possibly subtract smallest datetime from all values).
+                # TODO: could also return an extra boolean column is_nan which could provide predictive signal.
+                X_datetime[datetime_feature] = pd.to_datetime(X[datetime_feature], errors='coerce').fillna(self._fillna_map[datetime_feature])
+                # X_datetime[datetime_feature] = pd.to_timedelta(X_datetime[datetime_feature]).dt.total_seconds()
+                # Parse the date into lots of derived fields.
+                # Most of the pandas Series.dt properties are here, a few are omitted (e.g. is_month_start) if they can be inferred
+                # from other features.
+                for feature in self.features:
+                    X_datetime[datetime_feature + '.' + feature] = getattr(X_datetime[datetime_feature].dt, feature).astype(int)
+
+                for i in range(len(X_datetime)):
+                    X_datetime[datetime_feature + "_holiday"].iloc[i] = int(date(X_datetime[datetime_feature + ".year"].iloc[i], X_datetime[datetime_feature + ".month"].iloc[i], X_datetime[datetime_feature + ".day"].iloc[i]) in holidays_list)
+                X_datetime[datetime_feature] = pd.to_numeric(X_datetime[datetime_feature])
+
+                X_datetime["dow_hour"] = X_datetime[datetime_feature + ".dayofweek"] + X_datetime[datetime_feature + ".hour"] / 24
```

Comment:
This logic should be toggle-able. All of these things should have parameters that allow them to be enabled/disabled, as they are not always desirable.

Reply:
I set toggles that let users choose whether they use holidays and sincos encoding or not.